src/agreement/mod.rs

```diff
@@ -189,92 +223,152 @@ impl<NodeUid: Clone + Debug + Eq + Hash + Ord> Agreement<NodeUid> {
+        self.messages
+            .push_back(AgreementMessage::conf(self.epoch, v));
+        // Trigger the start of the `Conf` round.
+        self.conf_round = true;
+        // Receive the `Conf` message locally.
+        let our_uid = self.uid.clone();
+        self.handle_conf(&our_uid, v)
+    }
+
+    /// Waits until at least (N − f) `Aux` messages have been received, such that
+    /// the set of values carried by these messages, vals, are a subset of
+    /// bin_values (note that bin_values_r may continue to change as `BVal`
+    /// messages are received, thus this condition may be triggered upon arrival
+    /// of either an `Aux` or a `BVal` message).
     fn handle_aux(&mut self, sender_id: &NodeUid, b: bool) -> AgreementResult<()> {
+        // Perform the `Aux` message round only if a `Conf` round hasn't started yet.
+        if self.conf_round {
+            return Ok(());
+        }
         self.received_aux.insert(sender_id.clone(), b);
-        self.try_coin()
+        if self.bin_values == BinValues::None {
+            return Ok(());
+        }
+        if self.count_aux() < self.num_nodes - self.num_faulty_nodes {
+            // Continue waiting for the (N - f) `Aux` messages.
+            return Ok(());
+        }
+        // Start the `Conf` message round.
+        self.send_conf()
+    }
+
+    fn handle_conf(&mut self, sender_id: &NodeUid, v: BinValues) -> AgreementResult<()> {
+        self.received_conf.insert(sender_id.clone(), v);
+        self.try_finish_conf_round()
+    }
+
+    fn try_finish_conf_round(&mut self) -> AgreementResult<()> {
+        if self.conf_round {
+            let (count_vals, vals) = self.count_conf();
+            if count_vals < self.num_nodes - self.num_faulty_nodes {
+                // Continue waiting for (N - f) `Conf` messages
+                return Ok(());
+            }
+            self.invoke_coin(vals)
+        } else {
+            Ok(())
+        }
     }
 
     fn send_aux(&mut self, b: bool) -> AgreementResult<()> {
-        // Multicast AUX.
+        // Multicast `Aux`.
         self.messages
-            .push_back(AgreementMessage::Aux(self.epoch, b));
-        // Receive the AUX message locally.
+            .push_back(AgreementMessage::aux(self.epoch, b));
+        // Receive the `Aux` message locally.
         let our_uid = self.uid.clone();
         self.handle_aux(&our_uid, b)
     }
 
-    /// AUX_r messages such that the set of values carried by those messages is
-    /// a subset of bin_values_r. Outputs this subset.
+    /// The count of `Aux` messages such that the set of values carried by those messages is a
+    /// subset of bin_values_r.
     ///
-    /// FIXME: Clarify whether the values of AUX messages should be the same or
-    /// not. It is assumed in `count_aux` that they can differ.
+    /// FIXME: Clarify whether the values of `Aux` messages should be the same or not. It is assumed
+    /// in `count_aux` that they can differ.
```

Comment:
Can that `FIXME` be removed? The following paragraph seems to answer this question.